src/main/java/com/orbitz/consul/SessionClient.java

```diff
@@ -8,176 +10,202 @@
-      }
-
-      target = addParams(target.path("info").path(sessionId), QueryOptions.BLANK);
-
-      List<SessionInfo> sessionInfo = Arrays.asList(target
-              .request().accept(MediaType.APPLICATION_JSON_TYPE).get(SessionInfo[].class));
-
-      return sessionInfo != null && sessionInfo.size() != 0 ? Optional.of(sessionInfo.get(0)) : Optional.<SessionInfo>absent();
-   }
-
+    private static final GenericType<SessionCreatedResponse> SESSION_CREATED_RESPONSE_TYPE =
+            new GenericType<SessionCreatedResponse>() {
+            };
+    private static final GenericType<List<SessionInfo>> SESSION_INFO_LIST_TYPE =
+            new GenericType<List<SessionInfo>>() {
+            };
+
+    private final WebTarget webTarget;
+
+    /**
+     * Constructs an instance of this class.
+     *
+     * @param webTarget The {@link WebTarget} to base requests from.
+     */
+    SessionClient(WebTarget webTarget) {
+        this.webTarget = webTarget;
+    }
+
+    /**
+     * Retrieves the host/port of the Consul leader.
+     * 
+     * GET /v1/status/leader
+     *
+     * @return The host/port of the leader.
+     */
+    public String getLeader() {
+        return webTarget.path("leader").request().get(String.class)
+                .replace("\"", "").trim();
+    }
+
+    /**
+     * Create Session.
+     * 
+     * PUT /v1/session/create
+     *
+     * @param value The session to create.
+     * @return ID of the newly created session .
+     */
+    public SessionCreatedResponse createSession(final Session value) {
+        return createSession(value, null);
+    }
+
+    /**
+     * Create Session.
+     * 
+     * PUT /v1/session/create
+     *
+     * @param value The session to create.
+     * @param dc    The data center.
+     * @return Response containing the session ID.
+     */
+    public SessionCreatedResponse createSession(final Session value, final String dc) {
+        WebTarget target = webTarget;
+
+        if (dc != null) {
+            target = webTarget.queryParam("dc", dc);
+        }
+
+        SessionCreatedResponse session = target.path("create").request().put(Entity.entity(value,
+                MediaType.APPLICATION_JSON_TYPE), SESSION_CREATED_RESPONSE_TYPE);
+
+        return session;
+    }
+
+    public Optional<SessionInfo> renewSession(final String sessionId) {
+        return renewSession(null, sessionId);
+    }
+
+    /**
+     * Renews a session.
+     *
+     * @param dc        The datacenter.
+     * @param sessionId The session ID to renew.
+     * @return The {@link SessionInfo} object for the renewed session.
+     */
+    public Optional<SessionInfo> renewSession(final String dc, final String sessionId) {
+        WebTarget target = webTarget;
+
+        if (dc != null) {
+            target = webTarget.queryParam("dc", dc);
+        }
+
+        List<SessionInfo> sessionInfo = target.path("renew").path(sessionId).request().put(Entity.entity("{}",
```

Comment:
what happens if the session doesn't exist? I think it would end up with a serialization error, as consul would return a response like so:

```
$ http PUT localhost:8500/v1/session/renew/foo
HTTP/1.1 404 Not Found
Content-Length: 26
Content-Type: text/plain; charset=utf-8
Date: Mon, 19 Oct 2015 15:32:08 GMT

Session id 'foo' not found
```